library/goptions.mli

```diff
@@ -172,6 +172,14 @@ type option_value =
   | StringValue of string
   | StringOptValue of string option
 
+val set_option_value : ?locality:option_locality ->
+  ('a -> option_value -> option_value) -> option_name -> 'a -> unit
```

Comment:
I'd prefer if the return type of this function was `(unit, Pp.t) result` now that you are exporting it, but we can live with the current version if you don't feel like modifying it.

Reply:
>I'd prefer if the return type of this function was (unit, Pp.t) result now that you are exporting it

Not seeing the point. Note the exception is raised in the callback, not by set_value itself.

Reply:
Indeed you'd need to modify the callback too; anyways that's just "do if you want", but the mess with functions throwing exceptions is real.